sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeChallenge.java

```diff
@@ -0,0 +1,90 @@
+package com.azure.identity;
+
+import com.microsoft.aad.msal4j.DeviceCode;
+
+/**
+ * Response returned from the STS device code endpoint containing information necessary for
+ * device code flow
+ */
+public class DeviceCodeChallenge {
```

Comment:
The code structure in this class is a little strange. Can we rearrange it to put fields at top, constructor, and then methods? I also wonder if we both unwrapping DeviceCode or if it is just referenced as necessary in the getter code. Finally, add the `@Fluent` and `@Immutable` annotations to this class.